app/models/physical_server_provision_task/state_machine.rb

```diff
@@ -0,0 +1,36 @@
+module PhysicalServerProvisionTask::StateMachine
+  def run_provision
+    raise MiqException::MiqProvisionError, "Unable to find #{model_class} with id #{source_id.inspect}" if source.blank?
+    dump_obj(options, "MIQ(#{self.class.name}##{__method__}) options: ", $log, :info)
+    signal :start_provisioning
+  end
+
+  def start_provisioning
+    update_and_notify_parent(:message => msg('start provisioning'))
```

Comment:
@miha-plesko why go through the trouble of updating the parent message if you're just going to raise immediately after?  The way this is now you can't even do `super` in the child class to get the benefit of sharing the `update_and_notify_parent` calls across subclasses because it would just raise.

I'm going to merge as is because this case _shouldn't_ ever happen

Reply:
@agrare my thinking was to improve logs for unimplemented provider (e.g. Lenovo): first log says "start provisioning" then it explodes "NotImplemented" which kind of reads nicely as "start provisioning is not implemented".

However, I see your point - the piece of code is unusable as soon as one performs override from the provider. Followup here https://github.com/ManageIQ/manageiq/pull/18706, thanks for merging.